src/elements/NewFunctionPopUp/NewFunctionPopUp.js

```diff
@@ -13,20 +14,53 @@ import './newFunctionPopUp.scss'
 
 const NewFunctionPopUp = ({
   action,
+  functionsStore,
   setNewFunctionKind,
+  setNewFunctionName,
+  setNewFunctionTag,
   setFunctionsPanelIsOpen
 }) => {
+  const [data, setData] = useState({
+    name: '',
+    runtime: DEFAULT_RUNTIME,
+    tag: ''
+  })
   const [isPopUpOpen, setIsPopUpOpen] = useState(false)
-  const [selectedRuntime, setSelectedRuntime] = useState(DEFAULT_RUNTIME)
+  const [validation, setValidation] = useState({
+    isNameValid: true
+  })
   const newFunctionBtn = useRef(null)
 
   const closePopUp = () => {
     setIsPopUpOpen(false)
-    setSelectedRuntime(DEFAULT_RUNTIME)
+    setData({
+      name: '',
+      runtime: DEFAULT_RUNTIME,
+      tag: ''
+    })
+    setValidation({
+      isNameValid: true
+    })
+  }
+
+  const checkValidation = () => {
+    return Object.values(validation).find(value => value === false) ?? true
```

Comment:
I think this is equivalent to:
```suggestion
    return !Object.values(validation).includes(false)
```